react/README.md

```diff
@@ -4,16 +4,312 @@
 
 ## Table of Contents
 
-  1. [Basic Rules](#basic-rules) #TBD
-  1. [Naming](#naming) #TBD
-  1. [Declaration](#declaration) #TBD
-  1. [Alignment](#alignment) #TBD
-  1. [Quotes](#quotes) #TBD
-  1. [Spacing](#spacing) #TBD
-  1. [Props](#props) #TBD
-  1. [Parentheses](#parentheses) #TBD
-  1. [Tags](#tags) #TBD
-  1. [Methods](#methods) #TBD
-  1. [Ordering](#ordering) #TBD
+  1. [Basic Rules](#basic-rules)
+  1. [Naming](#naming)
+  1. [Declaration](#declaration)
+  1. [Alignment](#alignment)
+  1. [Quotes](#quotes)
+  1. [Spacing](#spacing)
+  1. [Props](#props)
+  1. [Parentheses](#parentheses)
+  1. [Tags](#tags)
+  1. [Methods](#methods)
+  1. [Ordering](#ordering)
+
+## Basic Rules
+
+  - Only include one React component per file.
+  - Always use JSX syntax.
+  - Do not use `React.createElement` unless you're initializing the app from a file that is not JSX.
+
+## Class vs React.createClass
+
+  - Use class extends React.Component unless you have a very good reason to use mixins.
+
+  ```javascript
+  // bad
+  const Listing = React.createClass({
+    render() {
+      return <div />;
+    }
+  });
+
+  // good
+  class Listing extends React.Component {
+    render() {
+      return <div />;
+    }
+  }
+  ```
+
+## Naming
+
+  - **Extensions**: Use `.jsx` extension for React components.
+  - **Filename**: Use PascalCase for filenames. E.g., `ReservationCard.jsx`.
+  - **Reference Naming**: Use PascalCase for React components and camelCase for their instances:
+    ```javascript
+    // bad
+    const reservationCard = require('./ReservationCard');
+
+    // good
+    const ReservationCard = require('./ReservationCard');
+
+    // bad
+    const ReservationItem = <ReservationCard />;
+
+    // good
+    const reservationItem = <ReservationCard />;
+    ```
+
+    **Component Naming**: Use the filename as the component name. For example, `ReservationCard.jsx` should have a reference name of `ReservationCard`. However, for root components of a directory, use `index.jsx` as the filename and use the directory name as the component name:
+    ```javascript
+    // bad
+    const Footer = require('./Footer/Footer.jsx')
+
+    // bad
+    const Footer = require('./Footer/index.jsx')
+
+    // good
+    const Footer = require('./Footer')
+    ```
+
+
+## Declaration
+  - Do not use displayName for naming components. Instead, name the component by reference.
```

Comment:
Add note for HOCs who do not get their `displayName`s automatically generated by Babel.